src/alignment/mod.rs

```diff
@@ -229,6 +231,8 @@ mod tests {
             score: 5,
             xstart: 3,
             ystart: 0,
+            xend: 10,
+            yend: 10,
```

Comment:
xlen was being populated with the total length of the x TextSlice (x.len()), rather than the length of x that was aligned. I added xend and yend to indicate where the alignment ended on each sequence, to mirror xstart and ystart.  Should we just drop xlen?